pkg/instrumentation/defaultinstrumentation.go

```diff
@@ -77,10 +77,10 @@ func getDefaultInstrumentation(agentConfig *adapters.CwaConfig, additionalEnvs m
 
 	// set protocol by checking cloudwatch agent config for tls setting
 	exporterPrefix := http
-	isApplicationSignalsEnabled := agentConfig != nil && agentConfig.GetApplicationSignalsConfig() != nil
+	isApplicationSignalsEnabled := agentConfig != nil && agentConfig.GetApplicationSignalsMetricsConfig() != nil
 
 	if isApplicationSignalsEnabled {
-		if agentConfig.GetApplicationSignalsConfig().TLS != nil {
+		if agentConfig.GetApplicationSignalsMetricsConfig().TLS != nil {
 			exporterPrefix = https
```

Comment:
This should be done in a separate PR, but we should also set `exporterPrefix = https` when `agentConfig.GetApplicationSignalsTracesConfig().TLS != nil` and only pass in the environmental variables needed depending on the Application Signals configuration (e.g. `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` should only be set for traces.)